pkg/kcp/provider/gcp/vpcpeering/client/cloudComputeClient.go

```diff
@@ -59,11 +82,7 @@ func (c *networkClient) CreateVpcPeeringConnection(ctx context.Context, name *st
 		},
 	}
 
-	peeringOperationFromKyma, err := c.cnc.AddPeering(ctx, peeringRequestFromKyma)
-	if err != nil {
-		return nil, err
-	}
-	err = peeringOperationFromKyma.Wait(ctx)
+	_, err = c.cnc.AddPeering(ctx, peeringRequestFromKyma)
```

Comment:
This causes a new issue. You are ignoring the returned operation object, which means if your request succeeds (err != nil) but your operation eventually fails, you don't know why it failed and you can't update the object.
You may also think that you succeeded because you are not checking that operation is Done without any errors.